batchflow/models/torch/base_mixins.py

```diff
@@ -221,6 +222,197 @@ class OptimalBatchSizeMixin:
+            batch_size_estimation = (low + high) // 2
+
+        # then run precise method in neighbourhood of batch_size_estimation
+        return self._run_binary_scaling(inputs=inputs, targets=targets,
+                                        start_batch_size=batch_size_estimation,
+                                        low=low, high=high, method=method,
+                                        max_iters=max_iters, tail_size=tail_size,
+                                        max_memory=max_memory, pbar=pbar,
+                                        frequency=frequency)
+
+
+    def _run_power_scaling(self, inputs=None, targets=None, factor=2,
+                           start_batch_size=4, method='train', pbar='n',
+                           tail_size=10, max_memory=100, frequency=0.05):
+        """ Returns `batch_size` and `batch_size * factor`, so that `batch_size` 
+            fits in max_memory, while `batch_size * factor` does not."""
+
+        batch_size = start_batch_size if isinstance(start_batch_size, list) else [start_batch_size]
+        notifier = Notifier(n_iters=None, bar=pbar,
+                            monitors=[{'source': batch_size, 'name': 'batch_size'}])
+        while True:
+            try:
+                notifier.update()
+                input = inputs or self.make_placeholder_data(batch_size[-1], to_device=False)
+                input = list(input) if isinstance(input, (tuple, list)) else [input]
+                input = [item[:batch_size[-1]] for item in input]
+
+                with GPUMemoryMonitor(frequency=frequency) as monitor:
+                    if method == 'train':
+                        target = targets or self.predict(inputs=input, outputs='predictions')
+                        target = list(target) if isinstance(target, (tuple, list)) else [target]
+                        target = [item[:batch_size[-1]] for item in target]
+
+                        _ = self.train(inputs=input, targets=target, microbatch_size=False)
+                    else:
+                        _ = self.predict(inputs=input, microbatch_size=False)
+
+                data = monitor.data
+                consumed_memory = np.mean(np.sort(data, axis=0)[-tail_size:])
+
+                # exit if max_memory was exceeded
+                if consumed_memory > max_memory:
+                    batch_size.append(batch_size[-1] // factor)
+                    break
+
+                batch_size.append(batch_size[-1] * factor)
+            except RuntimeError as exception:
+                if self.is_oom_error(exception):
+                    batch_size.append(batch_size[-1] // factor)
+                    break
+                raise # some other error not memory related
+
+            self.garbage_collection_cuda()
+        self.garbage_collection_cuda()
+        return batch_size[-1], batch_size[-1] * factor
+
+
+    def _run_binary_scaling(self, inputs=None, targets=None, low=2, high=None,
```

Comment:
function is pretty much identical to the previous one: only the way to compute the next tested batch size differs